src/main/java/com/iota/iri/storage/LocalSnapshotsPersistenceProvider.java

```diff
@@ -0,0 +1,75 @@
+package com.iota.iri.storage;
+
+import com.iota.iri.utils.Pair;
+
+import java.util.List;
+
+/**
+ * Abstraction for localsnapshots-db persistence provider.
+ */
+public class LocalSnapshotsPersistenceProvider {
```

Comment:
Why doesn't this thing have an interface?
Shouldn't it implement `SnpashotsProvider`

If there are method you don't want to implement then you can return `UnsupportedOperationException`

Reply:
By the way an interesting design choice, I think a good one as well.
For example, currently `Tangle` lets us access data like Addresses which are not transactions, which I am not sure is so good.

Separating concerns is in general a good thing :-)